addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js

```diff
@@ -1510,7 +1510,7 @@ export class OdooEditor extends EventTarget {
         // Rejoin blocks that extractContents may have split in two.
         while (
             doJoin &&
-            next &&
+            next && !joinWith.textContent.includes(next.textContent) &&
```

Comment:
So it only merges nodes if the text content of one includes the text content of the other ? Isn't that a weird condition to have ? I don't understand this change. What role does this change play in fixing the issue ?

Reply:
@dmo-odoo This condition makes sure that when the destination(i.e. next) text is merged with our source(i.e. joinWith) text, then there is no further need to perform unnecessary deletion.

Reply:
I'm not sure I understand. Do you mean that it is used to test whether the merging already happened ?

Reply:
Yes, It is used to test that merging already happened and stops the further deletion.

Reply:
I have one more fix in my mind, If we remove endIsStart from this ( https://github.com/odoo/odoo/blob/15.0/addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js#L1490 ) line then also our problem is fixed.